tests/TestCase/ORM/LazyLoadEntityTraitTest.php

```diff
@@ -97,6 +100,29 @@ public function testMissingPrimaryKey()
         $this->assertNull($comment->author);
     }
 
+    /**
+     * tests that no db queries are performed if there's no association
+     *
+     * @return void
+     */
+    public function testNoDbQueriesOnEmptyAssociation()
+    {
+        $article = $this->Articles->newEntity([
+            'title' => 'Article with no author',
+            'body' => 'Article content',
+        ]);
+
+        Log::setConfig('queries', ['className' => 'Array']);
+        $this->connection->enableQueryLogging();
+
+        $this->assertFalse(isset($article->author));
+
+        $this->connection->disableQueryLogging();
+
+        $msgs = Log::engine('queries')->read();
+        $this->assertCount(3, $msgs);
```

Comment:
This test isn't super clear. Perhaps count the queries at the top (fixture inserts) and assert they don't change by the bottom.

Reply:
They will change because of:
```
 (int) 0 => 'debug connection=test duration=0 rows=2 PRAGMA table_info("authors")',
 (int) 1 => 'debug connection=test duration=0 rows=0 PRAGMA index_list("authors")',
 (int) 2 => 'debug connection=test duration=0 rows=0 PRAGMA foreign_key_list("authors")'
```

Reply:
I fixed this in https://github.com/jeremyharris/cakephp-lazyload/pull/27/commits/fd2990654494579eada9f16de1f87b27c3cc0f4f.